infrastructure-playbooks/cephadm-adopt.yml

```diff
@@ -233,26 +233,38 @@
       run_once: true
       delegate_to: '{{ groups[mon_group_name][0] }}'
 
-    - name: generate cephadm ssh key
-      command: "{{ ceph_cmd }} cephadm generate-key"
-      changed_when: false
-      run_once: true
-      delegate_to: '{{ groups[mon_group_name][0] }}'
-
-    - name: get the cephadm ssh pub key
-      command: "{{ ceph_cmd }} cephadm get-pub-key"
-      changed_when: false
-      run_once: true
-      register: cephadm_pubpkey
-      delegate_to: '{{ groups[mon_group_name][0] }}'
+    - name: Configure root ssh key for remote connections
+      block:
+        - name: generate cephadm ssh key
+          command: "{{ ceph_cmd }} cephadm generate-key"
+          changed_when: false
+          run_once: true
+          delegate_to: '{{ groups[mon_group_name][0] }}'
+    
+        - name: get the cephadm ssh pub key
+          command: "{{ ceph_cmd }} cephadm get-pub-key"
+          changed_when: false
+          run_once: true
+          register: cephadm_pubpkey
+          delegate_to: '{{ groups[mon_group_name][0] }}'
+      when: cephadm_ssh_user = 'root'
 
-    - name: allow cephadm key for {{ cephadm_ssh_user | default('root') }} account
+    - name: Use custom user with sudo rights instead of root for remote connections
+      block:
+        - name: add custom user private key
+          command: "{{ ceph_cmd }} ceph config-key set mgr/cephadm/ssh_identity_key -i /home/{{ cephadm_ssh_user }}/.ssh/id_rsa"
+    
+        - name: add custom user public key
+          command: "{{ ceph_cmd }} ceph config-key set mgr/cephadm/ssh_identity_pub -i /home/{{ cephadm_ssh_user }}/.ssh/id_rsa.pub"
+      when: cephadm_ssh_user != 'root'
```

Comment:
```suggestion
    - name: use custom user with sudo rights instead of root for remote connections
      when: cephadm_ssh_user != 'root'
      block:
        - name: add custom user private key
          command: "{{ ceph_cmd }} ceph config-key set mgr/cephadm/ssh_identity_key -i /home/{{ cephadm_ssh_user }}/.ssh/id_rsa"
    
        - name: add custom user public key
          command: "{{ ceph_cmd }} ceph config-key set mgr/cephadm/ssh_identity_pub -i /home/{{ cephadm_ssh_user }}/.ssh/id_rsa.pub"
```